### PR TITLE
Adding get_ceph_capacity function

### PR DESCRIPTION
**Pull request by Avilir**
This PR is to add function that retrieve the Storage capacity in GB to support Issue #1696

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -483,6 +483,26 @@ def get_ceph_status(self):
             "ceph status", out_yaml_format=False,
         )
 
+    def get_ceph_capacity(self):
+        """
+        The function gets the total storage capacity of the ocs cluster
+
+        Returns:
+            Total storage capacity in GiB (GiB is for development environment)
```
Comment (RazTamir) [inline ocs_ci/ocs/cluster.py]: str? int?

**Comment by mbukatov** [comment]
Also note that this capacity has nothing to do with space one can actually utilize.

File: ocs_ci/ocs/cluster.py (hunk 1 above)
Comment (Avilir) [reply ocs_ci/ocs/cluster.py]: fixed.

**Comment by Avilir** [comment]
> I see multiple problems with such functionality:
> 
>     * parsing of human readable plain text output
> 
>     * ceph storage utilization is a tricky concept, but the functions doesn't seem to indicate what it is reporting neither via name nor docstring
> 
> 
> I believe that for getting information about ceph storage utilisation, one should use `ct_pod.exec_ceph_cmd(ceph_cmd="ceph df")` which returns dictionary created from a yaml, where you can find:
> 
>     * `['stats']['total_bytes']`
> 
>     * `['stats']['total_used_bytes']`
> 
>     * `['stats']['total_used_raw_bytes']`
> 
> 
> and others.

ctualy, this output line is the summery of "ceph df", and it is more simple to get.

**Comment by Avilir** [comment]
> Also note that this capacity is not the space one can actually utilize, which depends on a ceph pool configuration, and also a little bit on a volume type.

I know that this is the total RAW capacity and not the usable capacity (which depend on the replica number).

**Comment by mbukatov** [comment]
> ctualy, this output line is the summery of "ceph df", and it is more simple to get.

I don't see how parsing human readable output via regexps is easier than asking for particular value from a python dictionary structure.

Also, Is the value you are returning `total_used_bytes` or `total_used_raw_bytes`?

**Comment by Avilir** [comment]
> > ctualy, this output line is the summery of "ceph df", and it is more simple to get.
> 
> I don't see how parsing human readable output via regexps is easier than asking for particular value from a python dictionary structure.
> 
> Also, Is the value you are returning `total_used_bytes` or `total_used_raw_bytes`?

Changed the return value to the : Usable Capacity (consider the replication).

**Comment by Avilir** [comment]
> I consider this way of getting ceph capacity as needlessly complex and prone to error.
> 
> That said, since the rest of the code in the same module is implemented this way, feel free to ignore me.

Changed to pull the information from the JSON output of 'ceph df'

File: ocs_ci/ocs/cluster.py [hunk 2]
```diff
@@ -483,6 +483,26 @@ def get_ceph_status(self):
             "ceph status", out_yaml_format=False,
         )
 
+    def get_ceph_capacity(self):
+        """
+        The function gets the total usable storage capacity of the ocs cluster.
```
Comment (annagitel) [inline ocs_ci/ocs/cluster.py]: did you mean returns instead of gets?

**Comment by shreekarSS** [comment]
For cluster expansion, we have this function 

def get_percent_used_capacity():
    ct_pod = pod_helpers.get_ceph_tools_pod()
    output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph df')
    total_used = (output.get('stats').get('total_used_raw_bytes'))
    total_avail = (output.get('stats').get('total_bytes'))
    return 100.0 * total_used / total_avail

**Comment by mbukatov** [comment]
Note that as stated in the docstring, this is an approximation, which may not work in all cluster configurations, use cases or in the future, but is probably fine in most cases.

**Pull request merged**